cas-server-support-saml-idp/src/main/java/org/jasig/cas/support/saml/web/idp/metadata/GenerateSamlIdpMetadata.java

```diff
@@ -0,0 +1,179 @@
+import net.shibboleth.idp.installer.metadata.MetadataGeneratorParameters;
+import net.shibboleth.utilities.java.support.security.SelfSignedCertificateGenerator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * The {@link GenerateSamlIdpMetadata} is responsible for
+ * generating metadata and required certificates for signing
+ * and encryption.
+ *
+ * @author Misagh Moayyed
+ * @since 4.3.0
+ */
+public final class GenerateSamlIdpMetadata {
+    private static final String URI_SUBJECT_ALTNAME_POSTFIX = "idp/metadata";
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    private final File metadataFile;
+
+    private final String entityId;
+    private final String scope;
+    private final String hostName;
+
+    private final File signingCertFile;
+    private final File signingKeyFile;
+
+    private final File encryptionCertFile;
+    private final File encryptionCertKeyFile;
+
+    /**
+     * Instantiates a new Generate saml metadata.
+     *
+     * @param metadataLocation the metadata location
+     * @param hostName the host name
+     * @param entityId the entity id
+     * @param scope the scope
+     */
+    public GenerateSamlIdpMetadata(final File metadataLocation, final String hostName,
+                                   final String entityId,
+                                   final String scope) {
+        this.entityId = entityId;
+        this.hostName = hostName;
+        this.scope = scope;
+
+        if (!metadataLocation.exists()) {
+            if (!metadataLocation.mkdir()) {
+                throw new IllegalArgumentException("Metadata location cannot be located/created");
+            }
+        }
+        this.metadataFile = new File(metadataLocation, "idp-metadata.xml");
+
+        this.signingCertFile = new File(metadataLocation, "idp-signing.crt");
+        this.signingKeyFile = new File(metadataLocation, "idp-signing.key");
+
+        this.encryptionCertFile = new File(metadataLocation, "idp-encryption.crt");
+        this.encryptionCertKeyFile = new File(metadataLocation, "idp-encryption.key");
+    }
+
+    public File getMetadataFile() {
+        return metadataFile;
+    }
+
+    public String getEntityId() {
+        return entityId;
+    }
+
+    public String getScope() {
+        return scope;
+    }
+
+    public String getHostName() {
+        return hostName;
+    }
+
+    public File getSigningCertFile() {
+        return signingCertFile;
+    }
+
+    public File getEncryptionCertFile() {
+        return encryptionCertFile;
+    }
+
+    public boolean isMetadataMissing() {
+        return !this.metadataFile.exists();
+    }
+
+    /**
+     * Generate the certificates for signing, encryption and then metadata.
+     */
+    public void generate() {
```

Comment:
Well it generates not just certificates by also metadata. What might be a better name to capture everything?